### PR TITLE
Add lint CI and first pass at resolving issues

### LINKED ISSUE #83: Add linting to CICD flow

### PR DESCRIPTION
**Pull request by elimillera**
Hey @bms63 here is the first go at linting, if something ends up to be too constraining, we can update that lintr. Just to make everyone aware, this will make other merges pretty nasty so I'm thinking we want to merge this in ASAP to make sure we don't have to duplicate work with merges.

I resolved most of the linting issues, there are some issues with comments in test and a function being overly complex.

closes #83

### CODE REVIEW

File: R/format.R [hunk 1]
```diff
@@ -39,7 +39,7 @@ xportr_format <- function(.df, metacore, domain = NULL, verbose = getOption("xpo
   if (!is.null(attr(.df, "_xportr.df_arg_"))) df_arg <- attr(.df, "_xportr.df_arg_")
```
Comment (bms63) [inline R/format.R:39]: Shall we increase the cyclomatic complexity to 20 or try and figure this one out?

**Review by averissimo — COMMENTED** [review]
I suggest adding some project options in `xportr.Rproj` as part of this PR:

This will help ensure we enforce some linter rules automatically:
* Removing trailing whitespace when saving a file lint rule: `trailing_whitespace_linter`
* New lines at the bottom of file `trailing_blank_lines_linter`

As well as avoid having native pipes when using the RStudio shortcut (Ctrl+Shift+M)

```
AutoAppendNewline: Yes
StripTrailingWhitespace: Yes

UseNativePipeOperator: No
```

File: R/format.R (hunk 1 above)
Comment (averissimo) [reply R/format.R:39]: Changing lines 76-84 to the code below should solve the problem.

The second loop is not really needed.

```R
for (i in seq_len(ncol(.df))) {
    format_sas <- purrr::pluck(format, colnames(.df)[i])
    if (is.na(format_sas) || is.null(format_sas))
      format_sas <- ""
    attr(.df[[i]], "format.sas") <- format_sas
  }
```

from:

```R
  for (i in names(format)) {
    attr(.df[[i]], "format.sas")  <- format[[i]]
  }
  
  # Convert NA formats to "" for haven
  for (i in seq_len(ncol(.df))) {
    if (is.na(attr(.df[[i]], "format.sas")) || is.null(attr(.df[[i]], "format.sas"))) 
      attr(.df[[i]], "format.sas") <- ""
  }
```

**Comment by averissimo** [comment]
@elimillera I made a modification to the branch to those 2 for loops to remove the warning _(per @bms63 suggestion on the call)_, this way we can keep the default values for the lint config.

I've also added the Rproj configuration to help to have a smooth experience and avoid those annoying trailing whitespace errors

**Pull request merged**